src/components/forms/textInput.stories.mdx

```diff
@@ -0,0 +1,224 @@
+    label={text("label", "Label")}
+    errorText={text("errorText", undefined)}
+    required={boolean("required", undefined)}
+    type={text("type", undefined)}
+    onChange={action("changed")}
+    IconRight={boolean("Show IconRight", undefined) ? CaretDownIcon : undefined}
+    IconLeft={boolean("Show IconLeft", undefined) ? CaretDownIcon : undefined}
+    before={text("before", undefined)}
+    after={text("after", undefined)}
+    orientation={options(
+      "orientation",
+      arrayToOptions(['vertical', 'horizontal']),
+      undefined,
+      { display: "inline-radio" }
+    )}
+  />
+</Story>
+
+<Preview isExpanded>
+  <>
+    <TextInput />
+    <br />
+    <TextInput
+      label="First Name"
+      required
+      errorText="Custom error text"
+      placeholder="Placeholder text"
+      IconRight={CaretDownIcon}
+      before="$"
+      after=".com"
+    />
+  </>
+</Preview>
+
+<Info type="info">
+  Additional props will be spread on the `input` element.
+</Info> 
+
+<Props of={TextInput} />
+
+## Example Layout
+
+<Preview>
+  <Story name="Example Layout">
+    <>
+      <Flex>
+        <FlexItem>
+          <TextInput label="First Name" required />
+        </FlexItem>
+        <FlexItem>
+          <TextInput label="Last Name" required />
+        </FlexItem >
+      </Flex>
+      <Flex>
+        <FlexItem>
+          <TextInput
+            type="email"
+            label="Email"
+            errorText="Enter a valid email address"
+            placeholder="e.g. anand@cbinsights.com"
+          />
+        </FlexItem>
+        <FlexItem>
+          <TextInput type="password" label="Password" />
+        </FlexItem>
+        <FlexItem>
+          <TextInput type="number" label="Zipcode" after="90210" />
+        </FlexItem>
+      </Flex>
+    </>
+  </Story>
+</Preview>
+
+### Throttle
+
+The prop `onThrottleChange` fires a callback that throttles less than 5 characters,
+and debounces 5 or more characters.
+https://www.peterbe.com/plog/how-to-throttle-and-debounce-an-autocomplete-input-in-react
+
+View this story in Canvas, in the action tab, to see how the callback is throttled.
+
+<Preview>
+  <Story name="Throttle">
+    <TextInput
+      onThrottledChange={action("changed")}
+    />
+  </Story>
+</Preview>
+
+
+### Controlled Input
+
+<Story name="Controlled Input">
+  <TextInput
+    value="Controlled"
+  />
+</Story>
+
+### Icons
+
+<Story name="Icons">
+  <TextInput
+    IconLeft={CaretDownIcon}
+    IconRight={CaretDownIcon}
+  />
+</Story>
+
+### Side label
+
+<Story name="Before & After">
+  <TextInput
+    before="before"
+    after="after"
+  />
+</Story>
+
+### Side Labels + Icon
+
+<Story name="Side + Icon">
+  <TextInput
+    label="Label"
+    placeholder="placeholder"
+    before="before"
+    after="after"
+    IconLeft={CaretDownIcon}
+    IconRight={CaretDownIcon}
+  />
+</Story>
+
+### Form management library
+You should be able to use this input in any form management library you'd like:
```

Comment:
I'd like to declare our preference for react-hook-form here

```suggestion
You may use this input in any form management library you'd like. We recommend `react-hook-form`, which is used in the example below:
```

Reply:
thanks for the suggested change 👍